app/client/src/components/pages/Community.Tabs.Monitoring.js

```diff
@@ -0,0 +1,1189 @@
+
+                {item.monitoringType === 'Sample Location' && (
+                  <WaterbodyInfo
+                    type="Sample Location"
+                    feature={feature}
+                    services={services}
+                  />
+                )}
+
+                <ViewOnMapButton feature={feature} />
+              </div>
+            </AccordionItem>
+          );
+        })}
+      </AccordionList>
+    );
+  }
+}
+
+function MonitoringTab({ monitoringDisplayed, setMonitoringDisplayed }) {
+  const services = useServicesContext();
+
+  const {
+    huc12,
+    monitoringGroups,
+    monitoringLocations,
+    monitoringLocationsLayer,
+    setMonitoringGroups,
+    watershed,
+  } = useContext(LocationSearchContext);
+
+  const [displayedMonitoringLocations, setDisplayedMonitoringLocations] =
+    useState([]);
+
+  const [allToggled, setAllToggled] = useState(true);
+
+  const [charGroupFilters, setCharGroupFilters] = useState('');
+
+  const [totalMeasurements, setTotalMeasurements] = useState(0);
+
+  const [totalSamples, setTotalSamples] = useState(0);
+
+  const [sortBy, setSortBy] = useState('locationName');
+
+  const drawMap = useCallback(
+    (monitoringLocationTogglesParam) => {
+      if (!monitoringLocationsLayer) return;
+
+      const addedStationUids = [];
+      let tempDisplayedMonitoringLocations = [];
+      let allOthersToggled = true;
+
+      for (let key in monitoringLocationTogglesParam) {
+        const group = monitoringLocationTogglesParam[key];
+        if (group.label === 'All') continue;
+
+        // if the location is toggled
+        if (group.toggled) {
+          group.stations.forEach((station) => {
+            // add the station to the display, if it has not already been added
+            if (!addedStationUids.includes(station.uniqueId)) {
+              addedStationUids.push(station.uniqueId);
+              tempDisplayedMonitoringLocations.push(station);
+            }
+          });
+        } else {
+          allOthersToggled = false;
+        }
+      }
+
+      setAllToggled(allOthersToggled);
+
+      plotStations(tempDisplayedMonitoringLocations, monitoringLocationsLayer);
+
+      if (tempDisplayedMonitoringLocations.length === 0) {
+        setDisplayedMonitoringLocations([]);
+        return;
+      }
+
+      setDisplayedMonitoringLocations(tempDisplayedMonitoringLocations);
+    },
+    [monitoringLocationsLayer],
+  );
+
+  // create the filter string for download links based on active toggles
+  const buildFilter = useCallback(
+    (selectedNames) => {
+      let filter = '';
+
+      let groups = {};
+      characteristicGroupMappings.forEach(
+        (mapping) => (groups[mapping.label] = mapping.groupNames),
+      );
+
+      for (const name of selectedNames) {
+        filter +=
+          '&characteristicType=' + groups[name].join('&characteristicType=');
+      }
+
+      setCharGroupFilters(filter);
+    },
+    [setCharGroupFilters],
+  );
```

Comment:
```suggestion
   const buildFilter = useCallback(
    (selectedNames, monitoringGroups) => {
      let filter = '';

      let groups = {};
      characteristicGroupMappings.forEach(
        (mapping) => (groups[mapping.label] = mapping.groupNames),
      );

      const groupsCount = Object.values(monitoringGroups).filter(
        (group) => group.label !== 'All',
      ).length;
      if (groupsCount !== selectedNames.length) {
        for (const name of selectedNames) {
          filter +=
            '&characteristicType=' + groups[name].join('&characteristicType=');
        }
      }

      setCharGroupFilters(filter);
    },
    [setCharGroupFilters],
  );
```

I was testing at http://localhost:3000/community/dc/monitoring. I noticed that when I downloaded the file, prior to changing the filters, I would get all of the records (379,820). If I toggled one of the switches off and back on again, I would get back a different number of records (). 

The issue is because the `charGroupFilter` is not fully accounting for everything in the `Other` category. I think you already noticed this issue. 

To fix this issue you just need to omit the `charGroupFilter` when all toggles are on. The above suggestion should fix the issue, just don't forget to pass `monitoringGroups` into all uses of `buildFilter`.

Reply:
Whoops, I didn't realize I could add the changes right here, but I just committed the suggestions, it looks to have done the trick, thanks!